NConfig/Implementation/Mergers/IMergeableConfigurationCollection.cs

```diff
@@ -0,0 +1,35 @@
+using System.Collections;
+using System.Configuration;
+
+namespace NConfig
+{
+    /// <summary>
+    ///   Mergeable configuration collection (still untyped).
+    ///   Is an IEnumerable(ConfigurationElement)
+    /// </summary>
+    public interface IMergeableConfigurationCollection : IEnumerable {

```

Comment:
I assume this interface should be implemented by user configuration code. If this is the case, this file shouldn't be inside "Implementation" folder, consider root, next to the other public interfaces.